tests/backup/backup_helper.go

```diff
@@ -1254,6 +1292,213 @@ func IsBackupLocationPresent(bkpLocation string, ctx context.Context, orgID stri
+func GetLatestScheduleBackupNameAfterCheck(ctx context.Context, scheduleName string, orgID string) (string, error) {
+	latestScheduleBackupName, err := GetLatestScheduleBackupName(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	_, err = backupSuccessCheck(latestScheduleBackupName, orgID, 0, 0, ctx)
+	if err != nil {
+		return "", err
+	}
+	return latestScheduleBackupName, nil
+}
+
+// GetOrdinalScheduleBackupUID returns the uid of the schedule backup at the specified ordinal position for the given schedule
+func GetOrdinalScheduleBackupUID(ctx context.Context, scheduleName string, ordinal int, orgID string) (string, error) {
+	allScheduleBackupUids, err := Inst().Backup.GetAllScheduleBackupUIDs(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupUids) == 0 {
+		return "", fmt.Errorf("no backups were found for the schedule [%s]", scheduleName)
+	}
+	if ordinal < 1 {
+		return "", fmt.Errorf("the provided ordinal value [%d] for schedule backups with schedule name [%s] is invalid. valid values range from 1 to [%d]", ordinal, scheduleName, len(allScheduleBackupUids))
+	}
+	if ordinal > len(allScheduleBackupUids) {
+		return "", fmt.Errorf("schedule backups with schedule name [%s] have not been created up to the provided ordinal value [%d]", scheduleName, ordinal)
+	}
+	return allScheduleBackupUids[ordinal-1], nil
+}
+
+// GetOrdinalScheduleBackupUIDAfterCheck returns the uid of the schedule backup at the specified ordinal position for the given schedule after success check
+func GetOrdinalScheduleBackupUIDAfterCheck(ctx context.Context, scheduleName string, ordinal int, orgID string) (string, error) {
+	ordinalScheduleBackupUid, err := task.DoRetryWithTimeout(func() (interface{}, bool, error) {
+		allScheduleBackupUids, err := Inst().Backup.GetAllScheduleBackupUIDs(ctx, scheduleName, orgID)
+		if err != nil {
+			return "", false, err
+		}
+		if len(allScheduleBackupUids) == 0 {
+			return "", true, fmt.Errorf("no backups were found for the schedule [%s]", scheduleName)
+		}
+		if ordinal < 1 {
+			return "", false, fmt.Errorf("the ordinal value [%d] for schedule backups with schedule name [%s] is invalid. valid values range from 1 to [%d]", ordinal, scheduleName, len(allScheduleBackupUids))
+		}
+		if ordinal > len(allScheduleBackupUids) {
+			return "", true, fmt.Errorf("schedule backups with schedule name [%s] have not been created up to the provided ordinal value [%d]", scheduleName, ordinal)
+		}
+		return allScheduleBackupUids[ordinal-1], false, nil
+	}, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second)
+	if err != nil {
+		return "", err
+	}
+	ordinalScheduleBackupName, err := Inst().Backup.GetBackupName(ctx, ordinalScheduleBackupUid.(string), orgID)
+	if err != nil {
+		return "", err
+	}
+	_, err = backupSuccessCheck(ordinalScheduleBackupName, orgID, 0, 0, ctx)
+	if err != nil {
+		return "", err
+	}
+	return ordinalScheduleBackupUid.(string), nil
+}
```

Comment:
Same here. Not required.

Reply:
You are right. I implemented this to maintain symmetry with the corresponding GetScheduleBackupName function.